pkg/git/repository.go

```diff
@@ -229,6 +247,41 @@ var clone = func(ctx context.Context, opts *CloneOptions) (*repo, error) {
 	return repo, nil
 }
 
+var createRepo = func(ctx context.Context, opts *CloneOptions) (string, error) {
+	host, orgRepo, _, _, _, _, _ := util.ParseGitUrl(opts.Repo)
+	providerType := opts.Provider
+	if providerType == "" {
+		u, err := url.Parse(host)
+		if err != nil {
+			return "", err
+		}
+
+		providerType = strings.TrimSuffix(u.Hostname(), ".com")
+	}
+
+	p, err := newProvider(&ProviderOptions{
+		Type: providerType,
+		Auth: &opts.Auth,
+		Host: host,
+	})
+	if err != nil {
+		return "", fmt.Errorf("Failed to create the repository: %w\nYou can try to manually create it before trying again.", err)
```

Comment:
It's usually bad to have a line break inside an error. Also errors should not start with capital letter and no ending punctuation, see [here](https://github.com/golang/go/wiki/CodeReviewComments#error-strings). I think the structure should be: `failed to create the repository, you can try to manually create it before trying again: %w`